clusterInfo.py

```diff
@@ -10,6 +10,8 @@ def __init__(self, name: str):
         self.name = name
         self.provision_host = ""
         self.network_api_port = ""
+        self.bmc_imc_hostnames = []  # type: list[str]
+        self.ipu_mac_addresses = []  # type: list[str]
```

Comment:
BMC_IMC is captured by self.bmcs. It should be stored in the same field in our DB

Reply:
Same comment for ipu_mac_addresses.

If you look at the template for an iso_cluster installation:

`clusters:
  - name : "iso_cluster"
    api_vip: "192.168.122.99"
    ingress_vip: "192.168.122.101"
    network_api_port: "eno12409"
    kind: "iso"
    install_iso: "${ISO}"
    masters:
    - name: "${NODE_NAME}"
      node: "localhost"
      type: "physical"
      bmc: "${BMC}"
      bmc_user: "root"
      bmc_password: "redhat"
      ip: "${IP}"
      mac: "${MAC}"
`

There are no new fields introduced

Reply:
I agree. We can definitely store them in the same columns that are already existing. I  think it makes sense to keep `self.bmc_imc_hostnames = []` and `self.ipu_mac_addresses = []` within CDA. This is because BMC value used by iso_cluster is different then the BMC value used by host deploy. So I think we should keep these fields to easily be able to tell the difference (but yes in the spreadsheet they should under the same column).

Reply:
just to elaborate on my previous comment: 
```clusters:
  - name : "ocpcluster"
    api_vip: "192.168.122.99"
    ingress_vip: "192.168.122.101"
    kubeconfig: "/root/kubeconfig.ocpcluster"
    version: "4.16.0-nightly"
    network_api_port: "eno12399"
    postconfig:
    - name: "dpu_operator_host"
    masters:
    - name: "nicmodecluster-master-1"
      type: "vm"
      node: "localhost"
      ip: "192.168.122.2"
    - name: "nicmodecluster-master-2"
      type: "vm"
      node: "localhost"
      ip: "192.168.122.3"
    - name: "nicmodecluster-master-3"
      type: "vm"
      node: "localhost"
      ip: "192.168.122.4"
    workers:
    - name: "${NODE_NAME}"
      type: "physical"
      node: "${HOSTNAME}"
      bmc_user: "root"
      bmc_password: "calvin"
      bmc: "${BMC_IP}"
```
the host does bmc_ip
```clusters:
  - name : "iso_cluster"
    api_vip: "192.168.122.99"
    ingress_vip: "192.168.122.101"
    network_api_port: "eno12409"
    kind: "iso"
    install_iso: "${ISO}"
    masters:
    - name: "${NODE_NAME}"
      node: "localhost"
      type: "physical"
      bmc: "${BMC}"
      bmc_user: "root"
      bmc_password: "redhat"
      ip: "${IP}"
      mac: "${MAC}"
    postconfig:
    - name: "rh_subscription"
      organization_id: "${ORGANIZATION_ID}"
      activation_key: "${ACTIVATION_KEY}"
    - name: "microshift"
    - name: "dpu_operator_dpu"
      rebuild_dpu_operators_images: false
```
and deploy_ipu also uses `BMC`. In this case it makes sense for us to separate the values out. Here we have self.bmcs will be the bmc for the host side and self.bmc_imc_hostnames = [] will be the bmc values for the ipu/microshift deployment.

Reply:
I don't understand why BMC of the host and BMC of the DPU is different. During the deployment of the IPU, there is no notion of a host. Hence there is no way to confuse the two.

If we follow the approach where we have `self.bmc_imc_hostnames`, then we will have self.bmc_some_other_vendor_hostnames`, .... for each DPU.

My understanding is that if you reuse the same fields for both the DPU and the host, I could reuse your code to deploy microshift on the host without needing more code. I know we don't want that, but this does show that by unifying the two, everything simplifies. Less code = less bugs = better.